src/Typesense.php

```diff
@@ -65,96 +69,130 @@ private function getOrCreateCollectionFromModel($model): Collection
     /**
      * @param $model
      *
-     * @return \Typesense\Collection
-     * @throws \Http\Client\Exception
      * @throws \Typesense\Exceptions\TypesenseClientError
+     * @throws \Http\Client\Exception
+     *
+     * @return \Typesense\Collection
      */
     public function getCollectionIndex($model): Collection
     {
         return $this->getOrCreateCollectionFromModel($model);
     }
 
     /**
-     * @param  \Typesense\Collection  $collectionIndex
-     * @param $array
+     * @param \Typesense\Collection $collectionIndex
+     * @param                       $array
      *
-     * @throws \Http\Client\Exception
+     * @throws \Typesense\Exceptions\ObjectNotFound
      * @throws \Typesense\Exceptions\TypesenseClientError
+     * @throws \Http\Client\Exception
+     *
+     * @return \Typesense\LaravelTypesense\Classes\TypesenseDocumentIndexResponse
      */
-    public function upsertDocument(Collection $collectionIndex, $array): void
+    public function upsertDocument(Collection $collectionIndex, $array): TypesenseDocumentIndexResponse
     {
         /**
          * @var $document Document
          */
-        $document = $collectionIndex->getDocuments()[$array['id']];
+        $document = $collectionIndex->getDocuments()[$array['id']] ?? null;
```

Comment:
I haven't tested this yet, but won't this re-introduce this issue? https://github.com/typesense/laravel-scout-typesense-driver/pull/8